src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1762 @@
+        private byte[] SerialWakeUp = new byte[] { 0x55, 0x55, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private byte[] I2cWakeUp = new byte[] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private int I2cMAxBuffer = 1024;
+        private ParametersFlags _parametersFlags;
+
+        private SpiDevice _spiDevice = null;
+        private I2cDevice _i2cDevice = null;
+        private GpioController _controller = null;
+        private SerialPort _serialPort = null;
+        private int _pin = 18;
+        private SecurityAccessModuleMode _securityAccessModuleMode = SecurityAccessModuleMode.Normal;
+        private uint _virtualCardTimeout = 0x17;
+
+        /// <summary>
+        /// Set or get the read timeout for I2C and SPI
+        /// Please refer to the documentation to set the right
+        /// timeout value depending on the communication
+        /// mode you are using
+        /// </summary>
+        public int ReadTimeOut { get; set; } = 500;
+
+        /// <summary>
+        /// The Log level
+        /// </summary>        
+        public LogLevel LogLevel { get { return LogInfo.LogLevel; } set { LogInfo.LogLevel = value; } }
+
+        /// <summary>
+        /// The location to log the info
+        /// </summary>
+        public LogTo LogTo { get { return LogInfo.LogTo; } set { LogInfo.LogTo = value; } }
+
+        /// <summary>
+        /// Firmware version information
+        /// </summary>
+        public FirmwareVersion FirmwareVersion { get; internal set; }
+
+        #region Spi and I2c Settings
+        /// <summary>
+        /// PN532 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 2_000_000;
+
+        /// <summary>
+        /// Only SPI Mode supported is Mode0
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// The default I2C address
+        /// </summary>
+        public const byte I2cDefaultAddress = 0x24;
+
+        #endregion        
+
+        /// <summary>
+        /// Create a PN532 using Serial Port
+        /// </summary>
+        /// <param name="portName">The port name</param>
+        /// /// <param name="logLevel">The log level</param>
+        public Pn532(string portName, LogLevel logLevel = LogLevel.None)
+        {
+            LogLevel = logLevel;
+
+            // Data bit : 8 bits,
+            // Parity bit : none,
+            // Stop bit : 1 bit,
+            // Baud rate : 115 200 bauds,
+            LogInfo.Log("Opening serial port 115200, Parity.None, 8 bits, StopBits.One", LogLevel.Debug);
+            _serialPort = new SerialPort(portName, 115200, Parity.None, 8, StopBits.One);
+            // Documentation page 39, serial timeout is 89 ms for 115 200
+            _serialPort.ReadTimeout = 89;
```

Comment:
How does this compare to ReadTimeout property on this class?

Reply:
As explained in the comment of ReadTimeout, it's used only for I2C and SPI. For serial port, the timeout is based on the communication speed with specific tables.